src/fsharp/FSharp.Core/prim-types.fs

```diff
@@ -2504,10 +2504,13 @@ namespace Microsoft.FSharp.Core
             let rec parse n acc = if n < l then parse (n+1) (acc *.. 2UL +.. (match s.Chars(n) with '0' -> 0UL | '1' -> 1UL | _ -> formatError())) else acc in          
             parse p 0UL
 
+        let inline removeUnderscores (s:string) =
+            s.Replace("_", "")
```

Comment:
I think this needs a null check to fix the tests.

    match s with
    | null -> s
    | _ -> s.Replace("_", "")

Same with other instance of this method in lex.fsl